### ISSUE TITLE
Should `S101` be ignored in a test file?

### ISSUE THREAD

**Issue by harupy** [charliermarsh/ruff#714]
Should `S101` (assert detection) be ignored in a test file? We do use `assert` in a test file:

```python
# tests/test_foo.py


def test_foo():
    assert foo() == 1
```

**Comment by harupy** [comment]
Another assert use case that should be considered ok:

```python
# tests/utils.py: A collection of helper functions for testing

def assert_foo(x, y):
   assert x == y


def assert_bar(x, y):
   assert x != y
```

**Comment by andersk** [comment]
It’s hard to imagine a heuristic that would satisfy everyone here. Some people might want to forbid `assert` even in tests, in favor of test-specific methods with more useful failure output like `self.assertEqual(x, y)`.

But better configurability would help. In general, as we add support for more opinionated rules, it does seem inevitable that people will want to be more selective about which files to apply them in. Perhaps we could take some inspiration from [ESLint’s `overrides` setting](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#how-do-overrides-work) and [mypy’s per-module options](https://mypy.readthedocs.io/en/stable/config_file.html#per-module-and-global-options)?

**Comment by harupy** [comment]
> But better configurability would help

Agree. It looks like `bandit` provides an option for specifying files where assert detection is disabled: 
https://github.com/PyCQA/bandit/blob/f909a7dea9495517a8550c9b9f6a75bd286ce7e3/bandit/plugins/asserts.py#L20

```
**Config Options:**
You can configure files that skip this check. This is often useful when you
use assert statements in test cases.
.. code-block:: yaml
    assert_used:
      skips: ['*_test.py', '*test_*.py']
```

**Comment by harupy** [comment]
Confirmed `--per-file-ignores=tests/*:S101` works.
